{"expand":"renderedFields,names,schema,operations,editmeta,changelog,versionedRepresentations","id":"12658842","self":"https://issues.apache.org/jira/rest/api/2/issue/12658842","key":"YARN-944","fields":{"issuetype":{"self":"https://issues.apache.org/jira/rest/api/2/issuetype/1","id":"1","description":"A problem which impairs or prevents the functions of the product.","iconUrl":"https://issues.apache.org/jira/secure/viewavatar?size=xsmall&avatarId=21133&avatarType=issuetype","name":"Bug","subtask":false,"avatarId":21133},"timespent":null,"project":{"self":"https://issues.apache.org/jira/rest/api/2/project/12313722","id":"12313722","key":"YARN","name":"Hadoop YARN","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/projectavatar?pid=12313722&avatarId=15135","24x24":"https://issues.apache.org/jira/secure/projectavatar?size=small&pid=12313722&avatarId=15135","16x16":"https://issues.apache.org/jira/secure/projectavatar?size=xsmall&pid=12313722&avatarId=15135","32x32":"https://issues.apache.org/jira/secure/projectavatar?size=medium&pid=12313722&avatarId=15135"},"projectCategory":{"self":"https://issues.apache.org/jira/rest/api/2/projectCategory/10292","id":"10292","description":"Scalable Distributed Computing","name":"Hadoop"}},"fixVersions":[],"aggregatetimespent":null,"resolution":null,"customfield_12312322":null,"customfield_12310220":"2013-07-20T01:21:19.116+0000","customfield_12312520":null,"customfield_12312323":null,"customfield_12312521":"Fri May 01 15:41:40 UTC 2015","customfield_12310420":"339035","customfield_12312320":null,"customfield_12310222":null,"customfield_12312321":null,"resolutiondate":null,"workratio":-1,"customfield_12312328":null,"customfield_12312329":null,"customfield_12312923":null,"customfield_12312326":null,"customfield_12312920":null,"customfield_12310300":null,"customfield_12312327":null,"customfield_12312921":null,"customfield_12312324":null,"customfield_12312720":null,"customfield_12312325":null,"lastViewed":null,"watches":{"self":"https://issues.apache.org/jira/rest/api/2/issue/YARN-944/watchers","watchCount":10,"isWatching":false},"created":"2013-07-19T21:43:07.746+0000","customfield_12310192":null,"customfield_12310191":null,"priority":{"self":"https://issues.apache.org/jira/rest/api/2/priority/3","iconUrl":"https://issues.apache.org/jira/images/icons/priorities/major.svg","name":"Major","id":"3"},"labels":[],"customfield_12312333":null,"customfield_12312334":null,"customfield_12313422":"false","customfield_12310310":"4.0","customfield_12312331":null,"customfield_12312332":null,"timeestimate":null,"aggregatetimeoriginalestimate":null,"customfield_12311120":null,"customfield_12312330":null,"versions":[{"self":"https://issues.apache.org/jira/rest/api/2/version/12324029","id":"12324029","description":"2.1.0-beta release","name":"2.1.0-beta","archived":false,"released":true,"releaseDate":"2013-08-22"}],"issuelinks":[{"id":"12384787","self":"https://issues.apache.org/jira/rest/api/2/issueLink/12384787","type":{"id":"10030","name":"Reference","inward":"is related to","outward":"relates to","self":"https://issues.apache.org/jira/rest/api/2/issueLinkType/10030"},"inwardIssue":{"id":"12701747","key":"YARN-1841","self":"https://issues.apache.org/jira/rest/api/2/issue/12701747","fields":{"summary":"YARN ignores/overrides explicit security settings","status":{"self":"https://issues.apache.org/jira/rest/api/2/status/5","description":"A resolution has been taken, and it is awaiting verification by reporter. From here issues are either reopened, or are closed.","iconUrl":"https://issues.apache.org/jira/images/icons/statuses/resolved.png","name":"Resolved","id":"5","statusCategory":{"self":"https://issues.apache.org/jira/rest/api/2/statuscategory/3","id":3,"key":"done","colorName":"green","name":"Done"}},"priority":{"self":"https://issues.apache.org/jira/rest/api/2/priority/3","iconUrl":"https://issues.apache.org/jira/images/icons/priorities/major.svg","name":"Major","id":"3"},"issuetype":{"self":"https://issues.apache.org/jira/rest/api/2/issuetype/1","id":"1","description":"A problem which impairs or prevents the functions of the product.","iconUrl":"https://issues.apache.org/jira/secure/viewavatar?size=xsmall&avatarId=21133&avatarType=issuetype","name":"Bug","subtask":false,"avatarId":21133}}}}],"customfield_12312339":null,"assignee":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=jianhe","name":"jianhe","key":"jianhe","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Jian He","active":true,"timeZone":"America/Los_Angeles"},"customfield_12312337":null,"customfield_12312338":null,"updated":"2015-05-01T15:41:40.482+0000","customfield_12312335":null,"customfield_12312336":null,"status":{"self":"https://issues.apache.org/jira/rest/api/2/status/1","description":"The issue is open and ready for the assignee to start work on it.","iconUrl":"https://issues.apache.org/jira/images/icons/statuses/open.png","name":"Open","id":"1","statusCategory":{"self":"https://issues.apache.org/jira/rest/api/2/statuscategory/2","id":2,"key":"new","colorName":"blue-gray","name":"To Do"}},"components":[],"timeoriginalestimate":null,"description":"The container is the AM container. It started and the AM failed during RM registration. The error message presented was about container launch failing.","customfield_10010":null,"timetracking":{},"customfield_12312026":null,"customfield_12312023":null,"customfield_12310320":null,"customfield_12312024":null,"customfield_12312340":null,"attachment":[{"self":"https://issues.apache.org/jira/rest/api/2/attachment/12593268","id":"12593268","filename":"nm.log","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=bikassaha","name":"bikassaha","key":"bikassaha","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?ownerId=bikassaha&avatarId=29845","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&ownerId=bikassaha&avatarId=29845","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&ownerId=bikassaha&avatarId=29845","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&ownerId=bikassaha&avatarId=29845"},"displayName":"Bikas Saha","active":true,"timeZone":"America/Los_Angeles"},"created":"2013-07-19T21:43:46.085+0000","size":138796,"mimeType":"text/plain","content":"https://issues.apache.org/jira/secure/attachment/12593268/nm.log"},{"self":"https://issues.apache.org/jira/rest/api/2/attachment/12593269","id":"12593269","filename":"rm.log","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=bikassaha","name":"bikassaha","key":"bikassaha","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?ownerId=bikassaha&avatarId=29845","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&ownerId=bikassaha&avatarId=29845","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&ownerId=bikassaha&avatarId=29845","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&ownerId=bikassaha&avatarId=29845"},"displayName":"Bikas Saha","active":true,"timeZone":"America/Los_Angeles"},"created":"2013-07-19T21:43:46.086+0000","size":99823,"mimeType":"text/plain","content":"https://issues.apache.org/jira/secure/attachment/12593269/rm.log"},{"self":"https://issues.apache.org/jira/rest/api/2/attachment/12594742","id":"12594742","filename":"YARN-944.patch","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=jianhe","name":"jianhe","key":"jianhe","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Jian He","active":true,"timeZone":"America/Los_Angeles"},"created":"2013-07-29T17:43:05.976+0000","size":3914,"mimeType":"text/plain","content":"https://issues.apache.org/jira/secure/attachment/12594742/YARN-944.patch"},{"self":"https://issues.apache.org/jira/rest/api/2/attachment/12593270","id":"12593270","filename":"yarn-site.xml","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=bikassaha","name":"bikassaha","key":"bikassaha","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?ownerId=bikassaha&avatarId=29845","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&ownerId=bikassaha&avatarId=29845","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&ownerId=bikassaha&avatarId=29845","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&ownerId=bikassaha&avatarId=29845"},"displayName":"Bikas Saha","active":true,"timeZone":"America/Los_Angeles"},"created":"2013-07-19T21:43:46.088+0000","size":4402,"mimeType":"text/xml","content":"https://issues.apache.org/jira/secure/attachment/12593270/yarn-site.xml"}],"aggregatetimeestimate":null,"customfield_12312341":null,"customfield_12312220":null,"customfield_12312022":null,"customfield_12310921":null,"customfield_12310920":"339355","customfield_12312823":null,"summary":"App failed with container launch failed even though container started","creator":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=bikassaha","name":"bikassaha","key":"bikassaha","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?ownerId=bikassaha&avatarId=29845","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&ownerId=bikassaha&avatarId=29845","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&ownerId=bikassaha&avatarId=29845","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&ownerId=bikassaha&avatarId=29845"},"displayName":"Bikas Saha","active":true,"timeZone":"America/Los_Angeles"},"subtasks":[],"customfield_12310291":null,"reporter":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=bikassaha","name":"bikassaha","key":"bikassaha","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?ownerId=bikassaha&avatarId=29845","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&ownerId=bikassaha&avatarId=29845","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&ownerId=bikassaha&avatarId=29845","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&ownerId=bikassaha&avatarId=29845"},"displayName":"Bikas Saha","active":true,"timeZone":"America/Los_Angeles"},"customfield_12310290":null,"aggregateprogress":{"progress":0,"total":0},"customfield_12311024":null,"environment":null,"customfield_12313520":null,"customfield_12311020":null,"duedate":null,"customfield_12310250":null,"progress":{"progress":0,"total":0},"comment":{"comments":[{"self":"https://issues.apache.org/jira/rest/api/2/issue/12658842/comment/13714115","id":"13714115","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=bikassaha","name":"bikassaha","key":"bikassaha","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?ownerId=bikassaha&avatarId=29845","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&ownerId=bikassaha&avatarId=29845","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&ownerId=bikassaha&avatarId=29845","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&ownerId=bikassaha&avatarId=29845"},"displayName":"Bikas Saha","active":true,"timeZone":"America/Los_Angeles"},"body":"Attaching RM/NM logs and yarn-site.xml","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=bikassaha","name":"bikassaha","key":"bikassaha","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?ownerId=bikassaha&avatarId=29845","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&ownerId=bikassaha&avatarId=29845","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&ownerId=bikassaha&avatarId=29845","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&ownerId=bikassaha&avatarId=29845"},"displayName":"Bikas Saha","active":true,"timeZone":"America/Los_Angeles"},"created":"2013-07-19T21:43:46.090+0000","updated":"2013-07-19T21:43:46.090+0000"},{"self":"https://issues.apache.org/jira/rest/api/2/issue/12658842/comment/13714119","id":"13714119","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=bikassaha","name":"bikassaha","key":"bikassaha","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?ownerId=bikassaha&avatarId=29845","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&ownerId=bikassaha&avatarId=29845","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&ownerId=bikassaha&avatarId=29845","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&ownerId=bikassaha&avatarId=29845"},"displayName":"Bikas Saha","active":true,"timeZone":"America/Los_Angeles"},"body":"Error shown on RM web UI for application. The application container actually started. So this message is wrong.\n{noformat}\nApplication application_1374261801151_0002 failed 2 times due to AM Container for appattempt_1374261801151_0002_000002 exited with exitCode: 1 due to: Exception from container-launch:\norg.apache.hadoop.util.Shell$ExitCodeException:\nat org.apache.hadoop.util.Shell.runCommand(Shell.java:458)\nat org.apache.hadoop.util.Shell.run(Shell.java:373)\nat org.apache.hadoop.util.Shell$ShellCommandExecutor.execute(Shell.java:578)\nat org.apache.hadoop.yarn.server.nodemanager.DefaultContainerExecutor.launchContainer(DefaultContainerExecutor.java:195)\nat org.apache.hadoop.yarn.server.nodemanager.containermanager.launcher.ContainerLaunch.call(ContainerLaunch.java:258)\nat org.apache.hadoop.yarn.server.nodemanager.containermanager.launcher.ContainerLaunch.call(ContainerLaunch.java:74)\nat java.util.concurrent.FutureTask$Sync.innerRun(FutureTask.java:303)\nat java.util.concurrent.FutureTask.run(FutureTask.java:138)\nat java.util.concurrent.ThreadPoolExecutor$Worker.runTask(ThreadPoolExecutor.java:886)\nat java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:908)\nat java.lang.Thread.run(Thread.java:662)\n.Failing this attempt.. Failing the application.\n{noformat}","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=bikassaha","name":"bikassaha","key":"bikassaha","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?ownerId=bikassaha&avatarId=29845","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&ownerId=bikassaha&avatarId=29845","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&ownerId=bikassaha&avatarId=29845","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&ownerId=bikassaha&avatarId=29845"},"displayName":"Bikas Saha","active":true,"timeZone":"America/Los_Angeles"},"created":"2013-07-19T21:45:05.213+0000","updated":"2013-07-19T21:45:05.213+0000"},{"self":"https://issues.apache.org/jira/rest/api/2/issue/12658842/comment/13714291","id":"13714291","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=ojoshi","name":"ojoshi","key":"ojoshi","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?ownerId=ojoshi&avatarId=16634","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&ownerId=ojoshi&avatarId=16634","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&ownerId=ojoshi&avatarId=16634","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&ownerId=ojoshi&avatarId=16634"},"displayName":"Omkar Vinit Joshi","active":true,"timeZone":"America/Los_Angeles"},"body":"By default resolve ip is set to true...\n{code}\n    boolean useIp = conf.getBoolean(\n      CommonConfigurationKeys.HADOOP_SECURITY_TOKEN_SERVICE_USE_IP,\n      CommonConfigurationKeys.HADOOP_SECURITY_TOKEN_SERVICE_USE_IP_DEFAULT);\n    setTokenServiceUseIp(useIp);\n{code}\ncan you try setting below parameter?\n{code}\n   <property>                                                                    \n     <name>yarn.resourcemanager.scheduler.address</name>                         \n     <value>localhost:54313</value>  <!-- your resource manager scheduler address -->                                            \n     <description>host is the hostname of the resourcemanager and port is the port\n       on which the Applications in the cluster talk to the Resource Manager.    \n     </description>                                                              \n   </property>\n{code}\n\nI don't know whether we need to fix this..","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=ojoshi","name":"ojoshi","key":"ojoshi","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?ownerId=ojoshi&avatarId=16634","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&ownerId=ojoshi&avatarId=16634","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&ownerId=ojoshi&avatarId=16634","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&ownerId=ojoshi&avatarId=16634"},"displayName":"Omkar Vinit Joshi","active":true,"timeZone":"America/Los_Angeles"},"created":"2013-07-20T01:21:19.116+0000","updated":"2013-07-20T01:21:19.116+0000"},{"self":"https://issues.apache.org/jira/rest/api/2/issue/12658842/comment/13716672","id":"13716672","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=sandyr","name":"sandyr","key":"sandyr","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Sandy Ryza","active":true,"timeZone":"America/Los_Angeles"},"body":"I'm facing this as well.  The error I'm getting on the AM side is\n\n{code}\n\n2013-07-23 10:58:33,426 ERROR [main] org.apache.hadoop.mapreduce.v2.app.rm.RMContainerAllocator: Exception while registering\norg.apache.hadoop.security.AccessControlException: SIMPLE authentication is not enabled.  Available:[TOKEN]\n        at sun.reflect.NativeConstructorAccessorImpl.newInstance0(Native Method)\n        at sun.reflect.NativeConstructorAccessorImpl.newInstance(NativeConstructorAccessorImpl.java:39)\n        at sun.reflect.DelegatingConstructorAccessorImpl.newInstance(DelegatingConstructorAccessorImpl.java:27)\n        at java.lang.reflect.Constructor.newInstance(Constructor.java:513)\n        at org.apache.hadoop.yarn.ipc.RPCUtil.instantiateException(RPCUtil.java:53)\n        at org.apache.hadoop.yarn.ipc.RPCUtil.unwrapAndThrowException(RPCUtil.java:104)\n        at org.apache.hadoop.yarn.api.impl.pb.client.ApplicationMasterProtocolPBClientImpl.registerApplicationMaster(ApplicationMasterProtocolPBClientImpl.java:109)\n        at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method)\n        at sun.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:39)\n        at sun.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:25)\n        at java.lang.reflect.Method.invoke(Method.java:597)\n        at org.apache.hadoop.io.retry.RetryInvocationHandler.invokeMethod(RetryInvocationHandler.java:175)\n        at org.apache.hadoop.io.retry.RetryInvocationHandler.invoke(RetryInvocationHandler.java:94)\n        at $Proxy29.registerApplicationMaster(Unknown Source)\n        at org.apache.hadoop.mapreduce.v2.app.rm.RMCommunicator.register(RMCommunicator.java:147)\n        at org.apache.hadoop.mapreduce.v2.app.rm.RMCommunicator.serviceStart(RMCommunicator.java:107)\n        at org.apache.hadoop.mapreduce.v2.app.rm.RMContainerAllocator.serviceStart(RMContainerAllocator.java:213)\n        at org.apache.hadoop.service.AbstractService.start(AbstractService.java:193)\n        at org.apache.hadoop.mapreduce.v2.app.MRAppMaster$ContainerAllocatorRouter.serviceStart(MRAppMaster.java:789)\n        at org.apache.hadoop.service.AbstractService.start(AbstractService.java:193)\n        at org.apache.hadoop.service.CompositeService.serviceStart(CompositeService.java:101)\n        at org.apache.hadoop.mapreduce.v2.app.MRAppMaster.serviceStart(MRAppMaster.java:1019)\n        at org.apache.hadoop.service.AbstractService.start(AbstractService.java:193)\n        at org.apache.hadoop.mapreduce.v2.app.MRAppMaster$1.run(MRAppMaster.java:1401)\n        at java.security.AccessController.doPrivileged(Native Method)\n        at javax.security.auth.Subject.doAs(Subject.java:396)\n        at org.apache.hadoop.security.UserGroupInformation.doAs(UserGroupInformation.java:1493)\n        at org.apache.hadoop.mapreduce.v2.app.MRAppMaster.initAndStartAppMaster(MRAppMaster.java:1397)\n        at org.apache.hadoop.mapreduce.v2.app.MRAppMaster.main(MRAppMaster.java:1330)\nCaused by: org.apache.hadoop.ipc.RemoteException(org.apache.hadoop.security.AccessControlException): SIMPLE authentication is not enabled.  Available:[TOKEN]\n        at org.apache.hadoop.ipc.Client.call(Client.java:1428)\n        at org.apache.hadoop.ipc.Client.call(Client.java:1381)\n        at org.apache.hadoop.ipc.ProtobufRpcEngine$Invoker.invoke(ProtobufRpcEngine.java:206)\n        at $Proxy28.registerApplicationMaster(Unknown Source)\n        at org.apache.hadoop.yarn.api.impl.pb.client.ApplicationMasterProtocolPBClientImpl.registerApplicationMaster(ApplicationMasterProtocolPBClientImpl.java:106)\n        ... 22 more\n{code}\n\nHaven't looked into this deeply, but it seems like this is caused by YARN-701?","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=sandyr","name":"sandyr","key":"sandyr","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Sandy Ryza","active":true,"timeZone":"America/Los_Angeles"},"created":"2013-07-23T18:11:30.824+0000","updated":"2013-07-23T18:11:30.824+0000"},{"self":"https://issues.apache.org/jira/rest/api/2/issue/12658842/comment/13717481","id":"13717481","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=hitesh","name":"hitesh","key":"hitesh","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Hitesh Shah","active":true,"timeZone":"America/Los_Angeles"},"body":"There are 2 different issues being reported in this jira: \n\n  - The first related to the underlying error which is due to the changes in YARN-701, default addresses of 0.0.0.0 combined with the use of ip in the tokens cause apps to fail. \n  - the second issue is related to what failure information is propagated back to the user. For any non-zero exit code, the user now sees a ShellExitCodeException - this change was done as part of YARN-814.\n\n \n","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=hitesh","name":"hitesh","key":"hitesh","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Hitesh Shah","active":true,"timeZone":"America/Los_Angeles"},"created":"2013-07-23T18:59:07.141+0000","updated":"2013-07-23T18:59:07.141+0000"},{"self":"https://issues.apache.org/jira/rest/api/2/issue/12658842/comment/13717579","id":"13717579","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=bikassaha","name":"bikassaha","key":"bikassaha","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?ownerId=bikassaha&avatarId=29845","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&ownerId=bikassaha&avatarId=29845","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&ownerId=bikassaha&avatarId=29845","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&ownerId=bikassaha&avatarId=29845"},"displayName":"Bikas Saha","active":true,"timeZone":"America/Los_Angeles"},"body":"Making this a blocker since its making debugging quite difficult by hiding the root cause and giving a potentially misleading error message.","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=bikassaha","name":"bikassaha","key":"bikassaha","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?ownerId=bikassaha&avatarId=29845","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&ownerId=bikassaha&avatarId=29845","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&ownerId=bikassaha&avatarId=29845","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&ownerId=bikassaha&avatarId=29845"},"displayName":"Bikas Saha","active":true,"timeZone":"America/Los_Angeles"},"created":"2013-07-23T20:34:07.992+0000","updated":"2013-07-23T20:34:07.992+0000"},{"self":"https://issues.apache.org/jira/rest/api/2/issue/12658842/comment/13722711","id":"13722711","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=jianhe","name":"jianhe","key":"jianhe","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Jian He","active":true,"timeZone":"America/Los_Angeles"},"body":"Today when MapReduce launches a container, Container errors are redirected to syserr file under container log directory, we are not able to view that in NM log. In case of container-launch error(the error after user process is launched), users now will always see ShellExitCodeException in NM log, and the detailed message can be retrieved from container log in web UI. YARN-814 fixed the case that the error before user process is launched will be logged both in NM log and container log.\n\nFor other applications on YARN that are not redirecting stderr, user can see the container errors both in NM log and container log.\n\nUploaded a patch to update the logging error message.","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=jianhe","name":"jianhe","key":"jianhe","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Jian He","active":true,"timeZone":"America/Los_Angeles"},"created":"2013-07-29T17:43:05.991+0000","updated":"2013-07-29T17:43:05.991+0000"},{"self":"https://issues.apache.org/jira/rest/api/2/issue/12658842/comment/13936354","id":"13936354","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=ozhurakousky","name":"ozhurakousky","key":"ozhurakousky","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Oleg Zhurakousky","active":true,"timeZone":"America/New_York"},"body":"Here is another way of reproducing it.\n{code}\npublic static void main(String[] args) throws Exception {\n\tConfiguration conf = new Configuration();\n\tApplicationMasterProtocol applicationsManager = ClientRMProxy.createRMProxy(conf, ApplicationMasterProtocol.class);\n\tRegisterApplicationMasterRequest request = RegisterApplicationMasterRequest.newInstance(\"\", 0, \"\");\n\tRegisterApplicationMasterResponse response = applicationsManager.registerApplicationMaster(request);\n}\n{code}\nExecution the above while connecting to the remote YARN (2.3.0) cluster results in:\n{code}\nCaused by: org.apache.hadoop.ipc.RemoteException(org.apache.hadoop.security.AccessControlException): SIMPLE authentication is not enabled.  Available:[TOKEN]\n\tat org.apache.hadoop.ipc.Client.call(Client.java:1406)\n       . . . .\n{code}\n\nLooking at the ipc Server code where the actual exception is being triggered I wonder, what is the rational for hardcoding the TOKEN as one of the authentication methods especially when SIMPLE is configured explicitly. ","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=ozhurakousky","name":"ozhurakousky","key":"ozhurakousky","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?avatarId=10452","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&avatarId=10452","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&avatarId=10452","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&avatarId=10452"},"displayName":"Oleg Zhurakousky","active":true,"timeZone":"America/New_York"},"created":"2014-03-16T00:51:18.554+0000","updated":"2014-03-16T00:51:18.554+0000"},{"self":"https://issues.apache.org/jira/rest/api/2/issue/12658842/comment/14523347","id":"14523347","author":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=djp","name":"djp","key":"djp","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?ownerId=djp&avatarId=16954","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&ownerId=djp&avatarId=16954","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&ownerId=djp&avatarId=16954","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&ownerId=djp&avatarId=16954"},"displayName":"Junping Du","active":true,"timeZone":"Asia/Shanghai"},"body":"[~jianhe], we have improved diagnostics info a lot since this JIRA get filed. Do you think we still need any improvement here? If not, how about resolve it as \"not a problem\"?","updateAuthor":{"self":"https://issues.apache.org/jira/rest/api/2/user?username=djp","name":"djp","key":"djp","avatarUrls":{"48x48":"https://issues.apache.org/jira/secure/useravatar?ownerId=djp&avatarId=16954","24x24":"https://issues.apache.org/jira/secure/useravatar?size=small&ownerId=djp&avatarId=16954","16x16":"https://issues.apache.org/jira/secure/useravatar?size=xsmall&ownerId=djp&avatarId=16954","32x32":"https://issues.apache.org/jira/secure/useravatar?size=medium&ownerId=djp&avatarId=16954"},"displayName":"Junping Du","active":true,"timeZone":"Asia/Shanghai"},"created":"2015-05-01T15:41:40.482+0000","updated":"2015-05-01T15:41:40.482+0000"}],"maxResults":9,"total":9,"startAt":0},"votes":{"self":"https://issues.apache.org/jira/rest/api/2/issue/YARN-944/votes","votes":0,"hasVoted":false},"worklog":{"startAt":0,"maxResults":20,"total":0,"worklogs":[]},"customfield_12311820":"0|i1mh5z:"}}